proto/usr/triton/bin/sdc-factoryreset

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/bash
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+#
+# Copyright (c) 2021, Joyent, Inc.
+#
+
+#
+# This is almost a verbatim copy of sdc-factoryreset from smartos-live.
+# The only change is to find the pool name via symink instead of asking smf.
+#
+
+#
+# Sets a ZFS property to perform a factory reset, that is, discards all state on
+# this machine and runs setup again.
+#
+
+set -o errexit
+set -o pipefail
+
+myname=$(basename $0)
+
+function abort()
+{
+	printf "\nFactory reset aborted.\n"
+	exit 1
+}
+
+function usage()
+{
+	printf "\nUsage: $myname [-h | --help]\n\n"
+	printf "Resets a machine to its originally installed state.  See "
+	printf "sdc-factoryreset(1)\nfor more information.\n"
+	exit 1
+}
+
+if [[ -n $1 ]] && [[ $1 = "--help" ]]; then
+	usage
+fi
+
+while getopts "h" opt
+do
+	case "$opt" in
+		h)	usage;;
+		*)	usage;;
+	esac
+done
+
+trap abort SIGINT
+
+printf "WARNING: This machine will reboot and destroy its ZFS pools after "
+printf "rebooting.\n"
+
+read -p "Do you want to proceed with the factory reset? (y/n) " -n 1
+
+if [[ $REPLY =~ ^[Yy]$ ]]; then
+	printf "\n\nThis will destroy ALL DATA on the system, including "
+	printf "potential customer data.\n"
+
+	read -p "Are you sure? (y/n) " -n 1
+
+	if [[ $REPLY =~ ^[Yy]$ ]]; then
+		printf "\n\nRebooting in 5 seconds ... "
+		sleep 5
+		printf "now!\n"
+
+		SYS_ZPOOL=$(basename "$(readlink /run/triton/system_pool)")
+		[[ -n ${SYS_ZPOOL} ]] || SYS_ZPOOL=zones
+
+		zfs set smartdc:factoryreset=yes ${SYS_ZPOOL}/var
```

Comment:
OKAY, so some of these ARE straight-ish ports from the SmartOS CN, so I guess using `smartdc` in some places isn't a bad idea or an oversight.

Reply:
Correct.

Particularly for sdc-factoryreset I chose to make the property the same so that if you specify  a reset under Linux and boot to SmartOS it'll still wipe the pool.

Currently SmartOS can't read Linux pools, but you can create the pool some properties disabled that should make it work. That's also something that's on my to do list. Regardless, I wanted the property to be compatible because whether we disable linux pool properties or pull in ZoL changes to illumos, eventually SmartOS and Linux will be able to read each other's pools.

Reply:
Also I believe (but haven't verified) that Linux will read SmartOS pools, so if you specify a reset on SmartOS then reboot the CN to Linux it will be able to do the wipe.

Reply:
Ok.